miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java

```diff
@@ -537,11 +540,15 @@ private void validateAliasOrThrow(Library library) throws IOException {
   }
 
   private void validateAliasUniqueness(Library library) throws IOException {
-    if (!namingScheme.duplicateLibraryAliasAllowed() && !listByAlias(library.getAlias()).isEmpty()) {
-      if (LimsUtils.isDetailedLibrary(library) && ((DetailedLibrary) library).hasNonStandardAlias()) {
-        // do nothing; nonstandard alias means duplicates are acceptable
-      } else {
-        // throw if duplicate aliases are not allowed (and in the case of a DetailedLibrary, if it has a standard alias
+    // duplicate aliases may be allowed via naming scheme, or with nonStandardAlias=true in the case of a DetailedLibrary
+    if (namingScheme.duplicateLibraryAliasAllowed()
+        || (LimsUtils.isDetailedLibrary(library) && ((DetailedLibrary) library).hasNonStandardAlias())) {
+      return;
+    }
+    List<Library> potentialDupes = listByAlias(library.getAlias());
+    for (Library potentialDupe : potentialDupes) {
+      if (library.getId() == AbstractLibrary.UNSAVED_ID || library.getId() != potentialDupe.getId()) {
```

Comment:
No. `DefaultSampleService#validateAliasUniqueness` is only called when a new sample is created or an existing sample's alias is changed